src/lib/isSlug.js

```diff
@@ -1,6 +1,7 @@
 import assertString from './util/assertString';
 
-let charsetRegex = /^[^\s-_](?!.*?[-_]{2,})[a-z0-9-\\][^\s]*[^-_\s]$/;
+let charsetRegex = /^(?![-_])^[^\s](?!.*?[-_]{2,})[a-z0-9*&\\]*(?:[-_][a-z0-9*&\\]+)*[^-_\s](?!\.)$/;
```

Comment:
Should 'a-a' be considered a slug? I ran this example with this regex and it is not highlighting.

Reply:
Yes, the string 'a-a' should be considered a slug @Zhuochengyu

Thank you for pointing out this exception. I have pushed another update for the regex and also included a new test case.